src/sv-pipeline/04_variant_resolution/scripts/shard_pysam.py

```diff
@@ -0,0 +1,34 @@
+import argparse
+from pysam import VariantFile
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('input_vcf')
+    parser.add_argument('shard_prefix')
+    parser.add_argument('records_per_shard')
+    args = parser.parse_args()
+
+    records_per_shard = int(args.records_per_shard)
+
+    record_idx = 0
+    current_shard = -1
+    out_vcf = None
+    input_vcf = VariantFile(args.input_vcf, 'r', threads=2)
+    for record in input_vcf:
+        shard = int(record_idx / records_per_shard)
+        if shard != current_shard:
+            if out_vcf is not None:
+                out_vcf.close()
+            out_vcf = VariantFile("{}.{:05d}.vcf.gz".format(args.shard_prefix, shard), 'w', header=input_vcf.header, threads=2)
+        current_shard = shard
+        out_vcf.write(record)
+        record_idx = record_idx + 1
+    out_vcf.close()
```

Comment:
I don't think this would happen in cleanvcf5, but if we wanted to use this script generally I think this will crash if there are 0 records (`out_vcf` is never initialized). I think it's also better to emit a single empty shard if there are no records, since our concat task fails when there are 0 shards.